### DISCUSSION TITLE
Receipt API missing fields in Etsy V3 from Etsy V2

### DISCUSSION THREAD

**Discussion by itsdarrylnorris** [etsy/open-api#859] [Q&A] [answered]
I am working on a migration between Etsy v2 to Etsy V3. The endpoints that I am working on are: 

- Etsy V2 endpoint: `get /shops/{shop_name}/receipts?includes=Transactions/MainImage,Country`
- Etsy V3 endpoint: `get /v3/application/shops/{shop_id}/receipts`

In Etsy V2, there were an order_id and transactions[].product_data.sku, and I still need these fields in Etsy V3. I have searched around other APIs to see if I can get that data from a different API, but I am not able to see to find the order id or the product SKU. 

Do you guys know if possible to get these values by calling a different API?

**Comment by Gareth064** [comment, marked as answer]
order_id = `recipt.receipt_id`

transactions[].product_data.sku = `receipt.transactions[].sku`

**Reply by itsdarrylnorris** [reply]
Thank you @Gareth064

**Reply by xinlinz** [reply]
In the past, there was a receipt.receipt_id and a receipt.order_id but now there is only receipt_id. https://www.etsy.com/developers/documentation/reference/receipt
https://developers.etsy.com/documentation/reference/#operation/getShopReceipt
How should we get the order_id? I checked some old orders and these two are different ids.

**Reply by Gareth064** [reply]
I can't tell ye why there might have been 2 different ID's in the old v2, but in v3 there is just receipt_id.

What functionality is missing by not having order_id, or are you just curious why it is no longer there?

**Reply by xinlinz** [reply]
It would definitely be good to know why it is no longer there. I am a solution provider so customers are bound to ask about it when they see a set of data missing with the new API.

**Comment by colesnic89** [comment]
The same problem with `receipts[].shipping_details`. Do we have some replacement to check delivery option?

In V3 we have `receipts[].shipments`, but there is no specific type of delivery for the delivery service. It will be ok to make request for each receipt to check shipping by `receipts[].shipments[].receipt_shipping_id` ?

**Comment by Gareth064** [comment]
It's inside the reciept.transaction section. Sorry on phone so can't give an example right now

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by hemesh** [reply]
> It's inside the reciept.transaction section. Sorry on phone so can't give an example right now Sent from Outlook for Android<https://aka.ms/AAb9ysg>
> […](#)
> ________________________________ From: colesnic89 ***@***.***> Sent: Tuesday, April 25, 2023 2:19:54 PM To: etsy/open-api ***@***.***> Cc: Gareth Doherty ***@***.***>; Mention ***@***.***> Subject: Re: [etsy/open-api] Receipt API missing fields in Etsy V3 from Etsy V2 (Discussion #859) The same problem with receipts[].shipping_details. Do we have some replacement to check delivery option? In V3 we have receipts[].shipments, but there is no specific type of delivery for the delivery service. It will be ok to make request for each receipt to check shipping by receipts[].shipments[].receipt_shipping_id ? — Reply to this email directly, view it on GitHub<[#859 (comment)](https://github.com/etsy/open-api/discussions/859#discussioncomment-5719959)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFKGDFZ25TWGBUNSBW73NL3XC7FPVANCNFSM6AAAAAAUV2XE7I>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Thanks for this, I can see the now the field you mean. if there is not a transaction present however, or multiple transactions, I'm not 100% sure which one to take compared to the old method.

**Reply by Gareth064** [reply]
There will always be 1 transaction at minimum. The transaction defines the product which has been sold.

As for when there are multiple, yes it may seem confusing,  but the shipping data will be the same across all transactions. I simply take the data from the first one every time.

I wonder if Etsy have put it on all transactions because they are planning an update to their service in the future, whereby you could select individual items on a receipt to cancel/refund, rather than at the entire receipt level.

**Comment by hemesh** [comment]
I've had this problem too and reached out to developer support. Original question (which I think relates to this) was: -

> On V2, we called FindAllShopReceipts ([https://www.etsy.com/developers/documentation/reference/receipt#method_findallshopreceipts](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.etsy.com%2Fdevelopers%2Fdocumentation%2Freference%2Freceipt%23method_findallshopreceipts&data=05%7C01%7Cmintsoft.development%40theaccessgroup.com%7C3fa2f88595494dfec8ae08db46d86f48%7C739195a1f5d64d9aac42a1dbb7c7413d%7C0%7C0%7C638181667631344677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=dAFmOc8B51eoxx5EiS%2BM%2Fkm5CAFWmHb0sDRN9UMxu%2Bo%3D&reserved=0)), and on the response the following available property: “response.results[receiptId].shipping_details.shipping_method” and this was always populated by the API with a value.
>
> On the API documentation (V3), we can see the new call is now getShopReceipts ([https://developer.etsy.com/documentation/reference/#operation/getShopReceipts](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.etsy.com%2Fdocumentation%2Freference%2F%23operation%2FgetShopReceipts&data=05%7C01%7Cmintsoft.development%40theaccessgroup.com%7C3fa2f88595494dfec8ae08db46d86f48%7C739195a1f5d64d9aac42a1dbb7c7413d%7C0%7C0%7C638181667631344677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=THO3JzSjADmrhuG%2FU3KUBT3uaMrYErbhMbN5jmGsvIY%3D&reserved=0)), and via the response, we can see that “shipping_details” is no longer present per receipt, so we are unable to determine the shipping_method. 
>  
> Can you please advise what field we should be looking at for this receipt as it is not documented on the migration guide? If it’s something missing on there, can you update the guide and point us to where this is please?

And response was: -

> Please note that in v3, the Shipping_Method will be NULL if there was no upgraded shipping selected, so you can infer that by it being NULL that it is Standard Shipping. You should have a shipping_profile_id for the shop so that you can look up the shipping info. 
>  
> We initially were going to just have the shipping_method as a string value for all shipping but it became apparent we needed to modify this approach and so the shipping_method is more of a description when a shipping upgrade has been selected.

I need to confirm this, but I think that means that **shipping_details** might be provided if the **shipping_method** is an upgrade, otherwise it should be treated as the text "Standard Shipping"

Also, the OpenAPI documentation would need to be updated to stipulate this business logic I believe.

**Comment by hemesh** [comment]
I've also noticed some other missing fields too: -

- shipping_details (moved to transactions)
- needs_gift_wrap (no idea where this is currently, looks like it's not supported now?)
- currency_code (moved to each price/cost)

FYI @colesnic89

**Reply by colesnic89** [reply]
@hemesh  We have `receipts[].gift_wrap_price` (maybe can be used as replacements for `needs_gift_wrap`)

`"gift_wrap_price": {
      "amount": 0,
      "divisor": 100,
      "currency_code": "GBP"
  },`

**Reply by Gareth064** [reply]
`needs_gift_wrapped` is now `is_gift` and is at the receipt level.
![image](https://user-images.githubusercontent.com/22307223/234851922-1783d3dd-dbbd-4786-bbc9-08796a395b91.png)
